components/CodeBlock.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import { css } from '@emotion/react';
+
+interface Props {
+  code: string;
+  reviewLines?: number[];
+  highlightLines?: number[];
+  onChangeHighlights?: (selects: number[]) => void;
+}
+
+const CodeBlock = ({
+  code,
+  reviewLines,
+  highlightLines,
+  onChangeHighlights,
+}: Props) => {
+  const [selects, setSelects] = React.useState(new Set<number>());
+
+  const jsonCode = JSON.stringify(code);
+  const lines = jsonCode.slice(1, -1).split('\\n');
+
+  const reviewLineSet = new Set<number>(reviewLines);
+  const highlightLineSet = new Set<number>(highlightLines);
+
+  const onReviewAction = (line: number) => {
+    const reviewMode = !!onChangeHighlights;
+    if (!reviewMode) return;
+    const newSet = new Set<number>();
+    newSet.add(line);
+    setSelects(newSet);
+    onChangeHighlights?.(Array.from(newSet.keys()).sort());
+  };
+
+  return (
+    <StyledBox>
+      {lines.map((line, index) => (
+        <Line key={index}>
+          <PreNumBox
+            isSelect={selects.has(index + 1)}
+            line={index + 1}
+            onReviewLine={onReviewAction}
+          />
+          <Code
+            isReview={reviewLineSet.has(index + 1)}
+            isHighlight={highlightLineSet.has(index + 1)}
+          >
+            {line}
+          </Code>
+        </Line>
+      ))}
+    </StyledBox>
+  );
+};
+
+export default CodeBlock;
+
+const StyledBox = styled.ul`
+  width: 100%;
+  user-select: none;
+`;
+
+const Line = styled.li`
+  margin-bottom: 4px;
+`;
+
+const Code = styled.div<{ isReview: boolean; isHighlight: boolean }>`
+  display: inline;
+  word-spacing: 8px;
+
+  ${(props) =>
+    props.isReview
+      ? css`
+          background: rgba(236, 128, 0, 0.11);
+          border-bottom: 2px solid #ec8000;
+        `
+      : css``}
```

Comment:
```suggestion
    props.isReview
      && css`
          background: rgba(236, 128, 0, 0.11);
          border-bottom: 2px solid #ec8000;
        `}
```
이렇게 해도 문제없나요?
그러면 && 쓰는게 조금 더 짧아질듯요

Reply:
조건문이 false 이면 css 스타일 코드안에 false 가 들어가 버리게 됩니다.